### DISCUSSION TITLE
证书能不能出详细点的教程，华为的这个key申请教程不咋懂

### DISCUSSION THREAD

**Discussion by faceman0814** [0xJacky/nginx-ui#625] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/6b5e28ba-9ece-4b78-a82e-94a1d01574d0)

**Comment by faceman0814** [comment]
然后就一直卡在证书申请那里

**Reply by faceman0814** [reply]
正在获取证书，请稍等...

2024/10/15 16:44:55 [INFO] [Nginx UI] 正在准备 Lego 的配置

2024/10/15 16:44:55 [INFO] [Nginx UI] ACME User: System Initial User, Email: 1002784867@qq.com, CA Dir: https://acme-v02.api.letsencrypt.org/directory

2024/10/15 16:44:55 [INFO] [Nginx UI] 正在创建客户端用于与 CA 服务器通信

2024/10/15 16:44:55 [INFO] [Nginx UI] 使用 DNS01 Challenge 提供商

2024/10/15 16:44:55 [INFO] [Nginx UI] 正在设置环境变量

2024/10/15 16:44:55 [INFO] [Nginx UI] 正在获取证书

2024/10/15 16:44:55 [INFO] [*.test.faceman.cn, test.faceman.cn] acme: Obtaining bundled SAN certificate

2024/10/15 16:44:56 [INFO] [*.test.faceman.cn] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/416499387377

2024/10/15 16:44:56 [INFO] [test.faceman.cn] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/416499387387

2024/10/15 16:44:56 [INFO] [*.test.faceman.cn] acme: Could not find solver for: dns-01

2024/10/15 16:44:56 [INFO] [test.faceman.cn] acme: Could not find solver for: tls-alpn-01

2024/10/15 16:44:56 [INFO] [test.faceman.cn] acme: Could not find solver for: http-01

2024/10/15 16:44:56 [INFO] [test.faceman.cn] acme: Could not find solver for: dns-01

2024/10/15 16:44:56 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/416499387377

2024/10/15 16:44:57 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/416499387387

2024/10/15 16:44:57 [INFO] [Nginx UI] 正在重载 Nginx

obtain certificate error: error: one or more domains had a problem: [*.test.faceman.cn] [*.test.faceman.cn] acme: could not determine solvers [test.faceman.cn] [test.faceman.cn] acme: could not determine solvers

**Reply by 0xJacky** [reply]
您好，有试过重启 nginx-ui 再试一次吗

**Reply by faceman0814** [reply]
> 您好，有试过重启 nginx-ui 再试一次吗

试过的

**Reply by 0xJacky** [reply]
我这两天有空测试一下，华为这个是新增的 lego 的实现

**Reply by faceman0814** [reply]
> 我这两天有空测试一下，华为这个是新增的 lego 的实现

好的

**Comment by 0xJacky** [comment, marked as answer]
我今天测了一下，目前发布的 Nginx UI 版本漏输出了部分关键错误日志，目前已经修复。修复后，我的得到了真实的错误信息：
<img width="1433" alt="截屏2024-10-18 下午11 35 58" src="https://github.com/user-attachments/assets/6c9d0fc8-b8f6-4f06-8c41-acbe740fdd8e">
根据 https://support.huaweicloud.com/intl/en-us/usermanual-iam/iam_05_0001.html，请在这个页面中获取 region

<img width="2342" alt="image" src="https://github.com/user-attachments/assets/b3a0e8aa-9c83-40fe-b506-7902934e576b">

然后填入

<img width="506" alt="image" src="https://github.com/user-attachments/assets/1a4b2b7c-5a92-46a3-b935-97ad4b603acf">

保存，即可重新申请。
